libbeat/common/event_test.go

```diff
@@ -399,49 +395,60 @@ func BenchmarkConvertToGenericEventStringPointer(b *testing.B) {
 		ConvertToGenericEvent(MapStr{"key": &val})
 	}
 }
-
-func TestDeDotJsonMap(t *testing.T) {
-	var actualJSONBody map[string]interface{}
-	var expectedJSONBody map[string]interface{}
-
-	absPath, err := filepath.Abs("./testdata")
-	assert.NotNil(t, absPath)
-	assert.Nil(t, err)
-
-	actualJSONResponse, err := ioutil.ReadFile(absPath + "/json_map_with_dots.json")
-	assert.Nil(t, err)
-	err = json.Unmarshal(actualJSONResponse, &actualJSONBody)
-	assert.Nil(t, err)
-
-	dedottedJSONResponse, err := ioutil.ReadFile(absPath + "/json_map_dedot.json")
-	assert.Nil(t, err)
-	err = json.Unmarshal(dedottedJSONResponse, &expectedJSONBody)
-	assert.Nil(t, err)
-
-	actualJSONBody = DeDotJSON(actualJSONBody).(map[string]interface{})
-
-	assert.Equal(t, expectedJSONBody, actualJSONBody)
-}
-
-func TestDeDotJsonArray(t *testing.T) {
-	var actualJSONBody []interface{}
-	var expectedJSONBody []interface{}
-
-	absPath, err := filepath.Abs("./testdata")
-	assert.NotNil(t, absPath)
-	assert.Nil(t, err)
-
-	actualJSONResponse, err := ioutil.ReadFile(absPath + "/json_array_with_dots.json")
-	assert.Nil(t, err)
-	err = json.Unmarshal(actualJSONResponse, &actualJSONBody)
-	assert.Nil(t, err)
-
-	dedottedJSONResponse, err := ioutil.ReadFile(absPath + "/json_array_dedot.json")
-	assert.Nil(t, err)
-	err = json.Unmarshal(dedottedJSONResponse, &expectedJSONBody)
-	assert.Nil(t, err)
-
-	actualJSONBody = DeDotJSON(actualJSONBody).([]interface{})
-
-	assert.Equal(t, expectedJSONBody, actualJSONBody)
+func TestDeDotJSON(t *testing.T) {
+	var tests = []struct {
+		input  []byte
+		output []byte
+		valuer func() interface{}
+	}{
```

Comment:
Thanks for converting this to a table test. It makes it much easier to compare input with output. It makes it a bit harder to write json then before but I think the benefits outweight here.